offix-offline/src/main/java/org/aerogear/offixoffline/OfflineSyncService.kt

```diff
@@ -24,30 +27,42 @@ class OfflineSyncService : IntentService("OfflineService") {
         val isNetworkAvail = Offline.isNetwork()
 
         val mutationList = OfflineList.getInstance().offlineArrayList
-        val callbackList = OfflineList.getInstance().callbacksList
         val apolloClient = OfflineList.apClient
 
+        /*
+          Create an object of ApolloCall.Callback
+         */
+        val apolloCallback = object : ApolloCall.Callback<Any>() {
+            override fun onFailure(e: ApolloException) {
+                Log.d("Extension Callback - ", "$e")
+            }
+
+            override fun onResponse(response: Response<Any>) {
+                Log.d(TAG, response.data().toString())
+            }
+        }
+
         mutationList.forEachIndexed { index, currentMutation ->
             if (isNetworkAvail) {
                 Log.d(TAG, "Network Connected :  $isNetworkAvail")
-                val currentCallback = callbackList[index]
-                apolloClient?.mutate(currentMutation)?.enqueue(currentCallback)
+                apolloClient?.mutate(currentMutation)?.enqueue(apolloCallback)
             } else {
                 Log.d(TAG, "Network Connected :  $isNetworkAvail")
             }
-
-            /* Remove callbacks and mutations
-            */
-            mutationList.clear()
-            callbackList.clear()
-
-            /* Delete the mutations stored in the database as they have already been replicated to the server
-               via the array list method having mutations and callbacks because the app was still in foreground.
-               (This service starts when the app is in foreground, so no need to take the database approach here.)
-            */
-            Offline.getDb()?.mutationDao()?.deleteAllMutations()
-            Log.d(TAG, " Size of database list : ${Offline.getDb()?.mutationDao()?.getAllMutations()?.size}")
         }
+
+        /*
+         Remove mutations
+        */
+        mutationList.clear()
+
+
+        /* Delete the mutations stored in the database as they have already been replicated to the server
+           via the array list method having mutations and callbacks because the app was still in foreground.
+           (This service starts when the app is in foreground, so no need to take the database approach here.)
+        */
+        Offline.getDb()?.mutationDao()?.deleteAllMutations()
```

Comment:
When we implement the db part, we will have to remove mutations sequentially instead of deleting them all at once.

Reply:
yes, i know that's why db is a better appraoch.

Reply:
yes, i know that's why db is a better appraoch. In the next PR will do that :)